### ISSUE TITLE
Reliable method to look up icon by name

### ISSUE THREAD

**Issue by keyboardsamurai** [PrideChung/FontAwesomeKit#53]
Since the method to look up an icon by name as discussed in #37 [is buggy](https://github.com/PrideChung/FontAwesomeKit/issues/37#issuecomment-122588819), it would be nice to have a reliable method for the lookup.

**Comment by PrideChung** [comment]
Before adding this feature, I'm wondering why you need it. I've never actually need this feature in my project. The reason I added this method is I want to add some debugging support. Do you want to save the icon's name as some kind of configuration and recreate that icon later? In that case I think I should implement NSCoding on `FAKIcon` so an icon can be archived.

**Comment by keyboardsamurai** [comment]
It's exactly that - I am configuring a list of labels by providing them with a serialized json feed from a server. Since I want the user to be able to pick an icon, I have to send the name of the icon over the wire somehow. It would be awesome having this functionality.

**Comment by davidck** [comment]
@keyboardsamurai second this, I also encountered the duplicate issue. Enumerating through [FAKFontAwesome allIcons] also do not work.  I have a configuration which holds which icon to user by name.

**Comment by GabrielCartier** [comment]
I also encountered this issue but in my case, I received the font awesome string from the server and had to do the translation to properly display the icon.

**Comment by top2tech** [comment]
+1, I have dozens of items, each has different icon. I put them in a JSON file (and updatable from server). name lookup will help greatly for this scenario. 
Thanks!

**Comment by top2tech** [comment]
Another library has name look up: https://github.com/alexdrone/ios-fontawesome 
But I want to stick to this library.

**Comment by PrideChung** [comment]
@top2tech Do you need to save any other property into the JSON file, like icon color and the size?

**Comment by GabrielCartier** [comment]
I guess the other properties can be used in other ways. If they have size and color, it's easy to actually create the icon properly knowing those. The main issue is the name lookup.

I don't think adding the other properties would be useful with the existing methods.

**Comment by davidck** [comment]
Yes, the issue is name lookup. I'm currently using the output as an attributed string and configure it separately, so no urgent need for icon color/size.  Not being able to lookup some of these icons is actually creeping up on me since our designer heavily relies on icons from FontAwesome.  I'm currently hardcoding these until it's fixed.

**Comment by GabrielCartier** [comment]
Same here, I have implemented a few of them, is there anything preventing a simple lookupMethod?

I can do it and send a PR if @PrideChung doesn't have the time.

**Comment by PrideChung** [comment]
@GabrielCartier I'm busy for baby sitting lately, you would do us a big favor if you can implement this.

**Comment by davidck** [comment]
@GabrielCartier Looking at FAKFontAwesome.m, I think the simplest way would be to ensure [FAKFontAwesome allIcons] has no duplicates and that it has correct mapping to its matching class method.

The long way would be to change the code scheme to match the Font Awesome web scheme.
i.e. <i class="fa fa-star-o"></i>

So instead of @"\uf006", I'd propose @"fa-star-o".

**Comment by davidck** [comment]
Perhaps we can create a separate lookup method to fetch the appropriate class method to perform selector.
ex:
`FAKFontAwesome *icon = [FAKFontAwesome iconWithLookup:@"fa-star-o" size:18.0f];`

**Comment by GabrielCartier** [comment]
Yea, that's what I was thinking. I'll write it today if I got time and do a PR
